lib/source/membership.rb

```diff
@@ -29,6 +29,9 @@ def merged_with(csv)
 
         pr = reconciler.reconciliation_data rescue abort($!.to_s)
         pr.each { |r| id_map[r[:id]] = r[:uuid] }
+      else
```

Comment:
Philosophical. I'm (rather obviously) not a complete purist on this, but an `else` in OO code is almost always always a very bad smell. The OO way is to replace pretty much all conditionals with polymorphism. My standard rule of thumb is to treat as very suspicious any `if` that isn't either a guard clause or part of a factory method (and even in the latter case, data-driven is usually better than a giant case statement)